android/java/org/chromium/chrome/browser/ntp/sponsored/NTPUtil.java

```diff
@@ -0,0 +1,269 @@
+                        bundle.putInt(SponsoredImageUtil.NTP_TYPE, ntpType);
+                        rewardsBottomSheetDialogFragment.setArguments(bundle);
+                        rewardsBottomSheetDialogFragment.setNewTabListener(newTabListener);
+                        rewardsBottomSheetDialogFragment.show(chromeActivity.getSupportFragmentManager(), "rewards_bottom_sheet_dialog_fragment");
+                        rewardsBottomSheetDialogFragment.setCancelable(false);
+
+                        sponsoredTab.updateBannerPref();
+                    }
+                });
+            
+                turnOnAdsButton.setOnClickListener(new View.OnClickListener() {
+                    @Override
+                    public void onClick(View view) {
+                        NTPUtil.turnOnAds();
+                        nonDistruptiveBannerLayout.setVisibility(View.GONE);
+
+                        sponsoredTab.updateBannerPref();
+                    }
+                });
+
+                switch(ntpType) {
+                    case SponsoredImageUtil.BR_OFF:
+                        bannerText.setText(chromeActivity.getResources().getString(R.string.get_paid_to_see_image));
+                        learnMoreText.setVisibility(View.VISIBLE);
+                        break;
+                    case SponsoredImageUtil.BR_ON_ADS_OFF:
+                        bannerText.setText(chromeActivity.getResources().getString(R.string.get_paid_to_see_image));
+                        learnMoreText.setVisibility(View.VISIBLE);
+                        break;
+                    case SponsoredImageUtil.BR_ON_ADS_OFF_BG_IMAGE:
+                        bannerText.setText(chromeActivity.getResources().getString(R.string.you_can_support_creators));
+                        turnOnAdsButton.setVisibility(View.VISIBLE);
+                        break;
+                    case SponsoredImageUtil.BR_ON_ADS_ON:
+                        bannerText.setText(chromeActivity.getResources().getString(R.string.you_are_getting_paid));
+                        learnMoreText.setVisibility(View.VISIBLE);
+                        break;
+                }
+            }
+        }, 1500);
+    }
+
+    public static NTPImage getNTPImage() {
+        SharedPreferences mSharedPreferences = ContextUtils.getAppSharedPreferences();
+
+        if (mSharedPreferences.getBoolean(BackgroundImagesPreferences.PREF_SHOW_SPONSORED_IMAGES, true)
+            && SponsoredImageUtil.getSponsoredImages().size() > 0
+            && mSharedPreferences.getInt(BackgroundImagesPreferences.PREF_APP_OPEN_COUNT, 0) == 1
+            && SponsoredImageUtil.getTabIndex() == 2
+            && BraveAdsNativeHelper.nativeIsLocaleValid(Profile.getLastUsedProfile())
+            && !BravePrefServiceBridge.getInstance().getSafetynetCheckFailed()
+            && ChromeFeatureList.isEnabled(BraveFeatureList.BRAVE_REWARDS)) {
+            SponsoredImage sponsoredImage = SponsoredImageUtil.getSponsoredImage();
```

Comment:
indentation is wrong here, please check the entire PR for consistent 4 spaces indentation https://source.android.com/setup/contribute/code-style#use-spaces-for-indentation

Reply:
also here is a link for the chromium style guide where it differs from above https://chromium.googlesource.com/chromium/src/+/master/styleguide/java/java.md

Reply:
I thought we were going to switch to using the c++ code for this stuff? We shouldn't be duplicating it in java